libgalois/include/katana/analytics/ClusteringImplementationBase.h

```diff
@@ -150,34 +156,36 @@ struct ClusteringImplementationBase {
    * the unique clusters.
    */
   template <typename EdgeWeightType>
-  void SumVertexDegreeWeight(Graph* graph, CommunityArray& c_info) {
+  void SumVertexDegreeWeight(Graph* graph, CommunityArray* c_info) {
     katana::do_all(katana::iterate(*graph), [&](GNode n) {
       EdgeTy total_weight = 0;
-      auto& n_degree_wt =
-          graph->template GetData<DegreeWeight<EdgeWeightType>>(n);
+      // Calculate all out-edge weight
       for (auto ii = graph->edge_begin(n); ii != graph->edge_end(n); ++ii) {
         total_weight +=
             graph->template GetEdgeData<EdgeWeight<EdgeWeightType>>(ii);
       }
-      n_degree_wt = total_weight;
-      c_info[n].degree_wt = total_weight;
-      c_info[n].size = 1;
+      graph->template GetData<DegreeWeight<EdgeWeightType>>(n) = total_weight;
+      // All community have the single node
```

Comment:
The comment is not clear. What do you mean communities have a single node?

Reply:
these are singleton communities at the very start

Reply:
The comment is still not clear.